src/public/themes/blog/assets/js/comment.js

```diff
@@ -1,56 +1,124 @@
+					break;
+				default:
+					// Generic error message for other cases
+					$("#error_message").show().html("An error occurred. Please try again later.");
+					break;
+			}
+			console.error("AJAX error:", textStatus, errorThrown); // Log for debugging
+		}
+	} catch (e) {
+		// Handle parsing errors
+		console.error("Error parsing AJAX error response:", e);
+		$("#error_message").show().html("An unexpected error occurred. Please contact support.");
+	}
+}
 
-	});
-	
+// checking author name - validation for author name
+function checkingAuthorName(name) {
+	const regex = /^[A-Z \'.-]{2,90}$/i;
+	return regex.test(name);
 }
 
-function formSuccess(response) { 
+$(function () {
+	$("#commentForm").on('submit', function (event) {
+		event.preventDefault(); // Prevent default form submission
 
- let msg = $.parseJSON(response);
- $("#commentForm")[0].reset();
- $("#success_message").fadeIn().html(msg.success_message);
- setTimeout(function(){
-    $("#success_message").fadeOut("slow");
- }, 2000 );
-}
+		let post_url = $(this).attr("action");
+		let request_method = $(this).attr("method");
+		let form_data = $(this).serialize();
 
-function formError() {
-	
-	$("#commentForm").html(
-		'<div class="alert alert-danger">Could not reach server, please try again later.</div>'
-	);
-}
+		// Perform validation
+		let comment = $("#comment").val();
+		let name = $("#name").val();
+		let email = $("#email").val();
 
-$(function() {
-	$("#commentForm").on('submit', function(event){
-	
-		let post_url = $(this).attr("action"); //get form action url
-		let request_method = $(this).attr("method"); //get form GET/POST method
-		let form_data = $(this).serialize(); //Encode form elements for submission
-	
-		if (event.isDefaultPrevented()) {
-			// handle the invalid form...
-			$("#error_message").show().html("All Fields are Required");
-	
+		// validation rules
+		let isValid = true;
+
+		// 1. Comment: non-empty 
+		if (comment.trim() === "") {
+			$("#comment").addClass("is-invalid");
+			$("#error_message").show().html("Please enter your comment.");
+			isValid = false;
+		} else {
+			$("#comment").removeClass("is-invalid");
+		}
+
+		// 2. Name: non-empty, full name format
+		if (name.trim() === "") {
+			$("#name").addClass("is-invalid");
+			$("#error_message").show().html("Please enter your full name.");
+			isValid = false;
+		} else if (!checkingAuthorName(name)) {
+			$("#name").addClass("is-invalid");
+			$("#error_message").show().html("Please enter a valid name (2-90 characters, starting with an uppercase letter, spaces, apostrophes, periods, and hyphens allowed).");
+			isValid = false;
 		} else {
+			$("#name").removeClass("is-invalid");
+		}
+
+		// 3. Email: non-empty, valid email format
+		if (email.trim() === "") {
+			$("#email").addClass("is-invalid");
+			$("#error_message").show().html("Please enter your email address.");
+			isValid = false;
+		} else if (!/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email)) { // Basic email validation
```

Comment:
Make sure the regex used here, which is vulnerable to super-linear runtime due to backtracking, cannot lead to denial of service.